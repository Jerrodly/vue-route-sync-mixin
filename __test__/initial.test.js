/**
 * Summary.
 *  main test file.
 *
 * Description
 *  This is the main test file
 *
 * @file   This main test file for class vue-route-sync-mixin.
 * @author teeeemoji.
 * @since 2020-1-5
 */
import {createLocalVue, mount} from '@vue/test-utils'
import VueRouter from 'vue-router'
import Comp from './test-material/component.vue'
import App from './test-material/app.vue'
import EmptyComp from './test-material/empty.vue'

describe('initial data from url', function () {
  jest.setTimeout(500)
  let wrapper = null
  beforeAll(() => {
    const localVue = createLocalVue()
    const routes = [{path: '/', component: EmptyComp}, {path: '/comp', component: Comp}]
    const router = new VueRouter({
      routes
    })
    localVue.use(VueRouter)
    wrapper = mount(App, {
      localVue,
      router,
      attachToDocument: true
    })
    wrapper.vm.$router.push({
      path: '/comp',
      query: {
        ps: '1__100',
        pn: '1__200',
        n: '2__ztc',
        c: '3__%7B"start"%3A"123456","end"%3A"654321"%7D',
        d: '3__%5B1,2,3,"four"%5D',
        u: '4__',
        s: '5__true'
      }
    })
  })

  afterAll(() => {
    wrapper.destroy()
    wrapper = null
  })

  it('initial undefined data', function () {
    const comp = wrapper.vm.$refs.comp
    expect(comp.und).toMatchSnapshot()
  })

  it('initial number data', function () {
    const comp = wrapper.vm.$refs.comp
    expect(comp.pagination.pageNum).toMatchSnapshot()
    expect(comp.pagination.pageSize).toMatchSnapshot()
  })

  it('initial string data', function () {
    const comp = wrapper.vm.$refs.comp
    expect(comp.name).toMatchSnapshot()
  })

  it('initial boolean data', function () {
    const comp = wrapper.vm.$refs.comp
    expect(comp.smoke).toMatchSnapshot()
  })

  it('initial array and object data', function () {
    const comp = wrapper.vm.$refs.comp
    expect(comp.createDate).toMatchSnapshot()
    expect(comp.departments).toMatchSnapshot()
  })
})
