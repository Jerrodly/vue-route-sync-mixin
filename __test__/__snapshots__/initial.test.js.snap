// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial data from url initial array and object data 1`] = `
Object {
  "end": "654321",
  "start": "123456",
}
`;

exports[`initial data from url initial array and object data 2`] = `
Array [
  1,
  2,
  3,
  "four",
]
`;

exports[`initial data from url initial boolean data 1`] = `true`;

exports[`initial data from url initial number data 1`] = `200`;

exports[`initial data from url initial number data 2`] = `100`;

exports[`initial data from url initial string data 1`] = `"ztc"`;

exports[`initial data from url initial undefined data 1`] = `undefined`;
